iosApp/iosAppTests/iosAppTests.swift

```diff
@@ -101,6 +101,40 @@ class iosAppTests: XCTestCase {
         contentController.disconnectMessenger()
     }
 
+    func testConnectionClosed() {
+        // Pull the deployment for use later.
+        var deployment: Deployment?
+        do {
+            deployment = try Deployment()
+        } catch {
+            XCTFail("\(error.localizedDescription)")
+        }
+        guard let deployment = deployment else {
+            XCTFail("Failed to pull the deployment config.")
+            return
+        }
+
+        // Create 4 new content controllers. We'll use these to trigger a connection closed event.
+        // Right now, there's a max number of 3 open sesssions that use the same token.
+        // We'll open 4 and confirm that an error occurs in the fourth attempt on the oldest client.
+        var controllers = [TestContentController]()
+        for _ in 1...4 {
+            controllers.append(TestContentController(deployment: deployment))
+        }
+        controllers[0].connectionClosed = XCTestExpectation(description: "Wait for the ConnectionClosedEvent")
+        for controller in controllers {
+            controller.startMessengerConnection()
+            delay(3)
```

Comment:
is this delay in seconds? If yes. I assume a 1 second delay is more than enough to have consistent behvaiour.

Reply:
The 3 seconds may be excessive, but in the event of some backend slowness, a larger delay may prevent any unintended failures.